python_modules/dagster-graphql/dagster_graphql/schema/backfill.py

```diff
@@ -331,17 +339,54 @@ def _get_records(self, graphene_info: ResolveInfo) -> Sequence[RunRecord]:
 
     def _get_partition_run_data(self, graphene_info: ResolveInfo) -> Sequence[RunPartitionData]:
         if self._partition_run_data is None:
-            self._partition_run_data = (
-                graphene_info.context.instance.run_storage.get_run_partition_data(
-                    runs_filter=RunsFilter(
-                        tags={
-                            BACKFILL_ID_TAG: self._backfill_job.backfill_id,
-                        }
+            # Storage layer not equipped to calculate partition status for ranged job backfills so
+            # we perform the calculation ad hoc here.
+            partition_set = self._get_partition_set(graphene_info)
+            if (
+                partition_set
+                and partition_set.backfill_policy
+                and partition_set.backfill_policy.max_partitions_per_run != 1
+            ):  # ranged backfills
+                self._partition_run_data = self._get_partition_run_data_for_ranged_job_backfill(
+                    graphene_info.context.instance, partition_set
+                )
+            else:
+                self._partition_run_data = (
+                    graphene_info.context.instance.run_storage.get_run_partition_data(
+                        runs_filter=RunsFilter(
+                            tags={
+                                BACKFILL_ID_TAG: self._backfill_job.backfill_id,
+                            }
+                        )
                     )
                 )
-            )
         return self._partition_run_data
 
+    def _get_partition_run_data_for_ranged_job_backfill(
+        self, instance: DagsterInstance, partition_set: ExternalPartitionSet
+    ) -> Sequence[RunPartitionData]:
+        partitions_def = partition_set.get_partitions_definition()
+        runs = instance.get_runs(
+            RunsFilter(tags=DagsterRun.tags_for_backfill_id(self._backfill_job.backfill_id))
+        )
+        return [
+            RunPartitionData(
+                run_id=run.run_id,
+                partition=key,
+                status=run.status,
+                start_time=None,
+                end_time=None,
```

Comment:
I'm not sure the UI uses these fields, but I wonder if this is `run.start_time` or something simple like that? Might be hard to track down these being None later

Reply:
They were set to `None` here because they are already set to `None` in some cases when using the storage layer implementation for non-asset jobs, and `DagsterRun` doesn't have start/end time. However I realized I could pull `run_records` from the instance instead of runs, which has the necessary info, so updated. Good call.